edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs

```diff
@@ -56,17 +56,18 @@ V1PersistentVolumeClaim ExtractPvc(Mount mount, IDictionary<string, string> labe
                     Requests = new Dictionary<string, ResourceQuantity>() { { "storage", new ResourceQuantity($"{this.persistentVolumeClaimSizeMb}Mi") } }
                 },
             };
+
             // prefer persistent volume name to storage class name, if both are set.
             if (this.persistentVolumeName.HasValue)
             {
-                this.persistentVolumeName.ForEach(volumeName => persistentVolumeClaimSpec.VolumeName = volumeName);
+                persistentVolumeClaimSpec.VolumeName = this.persistentVolumeName.OrDefault();
```

Comment:
nit: I think you can just call OrDefault without if block.

Reply:
The if block is needed as the overwrite depends on the condition. We don't want the volumename to be none when no value exists.

Reply:
The default value of "persistentVolumeClaimSpec.VolumeName" is null; that's why you can directly call this.persistentVolumeName.OrDefault(); or you should directly call this.persistentVolumeName.Value inside if block.

Reply:
We prefer none than null. :)

Reply:
persistentVolumeClaimSpec.VolumeName is in third-party lib, not our code.